internal/app/secretless/handlers/pg/protocol/ssl.go

```diff
@@ -0,0 +1,149 @@
+package protocol
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"net"
+)
+
+type options map[string]string
+
+type TLSConfigWrapper struct {
+	UseTLS bool
+	tls.Config
+	VerifyCaOnly bool
+	Options options
+}
+
+func ResolveTLSConfig(o options) (TLSConfigWrapper, error) {
```

Comment:
I'm not a fan of the word "resolve" in general (there are better alternatives) but in this case it seems inaccurate as well: "resolving" implies to me a process of looking up, via a network or db, as in "the DNS name could not be resolved."  

Here we simply have a pure function that takes options and returns a `TLSConfigWrapper`.  That is, we have a constructor.  Let's stick with the convention and name it `NewTLSConfigWrapper(options)`.

*This comment is independent of my above suggestion that `TLSConfigWrapper` itself should be renamed :)